test/unit/Cardano/Wallet/PackfileSpec.hs

```diff
@@ -0,0 +1,56 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Cardano.Wallet.PackfileSpec (spec) where
+
+import Prelude
+
+import qualified Data.ByteString.Lazy.Char8 as L8
+
+import Test.Hspec
+    ( Spec, describe, it, shouldBe )
+
+import Cardano.Wallet.Binary.Packfile
+
+-- version 1 header
+packFileHeader :: L8.ByteString
+packFileHeader = "\254CARDANOPACK\NUL\NUL\NUL\SOH"
+
+test1 :: L8.ByteString
+test1 = packFileHeader <> "\0\0\0\5hello\0\0\0"
+
+test2 :: L8.ByteString
+test2 = packFileHeader
+        <> "\0\0\0\11first block\0"
+        <> "\0\0\0\12second block"
+
+spec :: Spec
+spec = do
+    describe "Decoding pack file" $ do
+        it "should not decode junk" $ do
+            let decoded = decodePackfile "junkjunkjunkjunk"
+            decoded `shouldBe` Left MissingMagicError
+
+        it "should ensure pack file type" $ do
+            let decoded = decodePackfile "\254CARDANOYOLO\NUL\NUL\NUL\SOH"
+            decoded `shouldBe` Left WrongFileTypeError
+
+        it "should ensure pack file version" $ do
+            let decoded = decodePackfile "\254CARDANOPACK\NUL\NUL\NUL\2"
+            decoded `shouldBe` Left VersionTooNewError
+
+        it "should decode an empty pack file" $ do
+            decodePackfile packFileHeader `shouldBe` Right []
+
+        it "should decode a single blob" $ do
+            decodePackfile test1 `shouldBe` Right ["hello"]
+
+        it "should decode multiple blobs" $ do
+            decodePackfile test2 `shouldBe` Right ["first block", "second block"]
+
+        it "should not decode overly large blobs" $ do
+            let decoded = decodePackfile (packFileHeader <> "\64\0\0\0")
+            decoded `shouldBe` Left (BlobDecodeError "read block of size: 1073741824")
```

Comment:
Wrap here to conform to https://github.com/input-output-hk/cardano-wallet/wiki/Coding-Standards#limit-line-length-to-80-characters

Reply:
Fixed